packages/client/hmi-client/src/composables/project.ts

```diff
@@ -0,0 +1,224 @@
+import { IProject } from '@/types/Project';
+import { Component, Ref, readonly, shallowRef } from 'vue';
+import * as ProjectService from '@/services/project';
+import * as CodeService from '@/services/code';
+import * as ArtifactService from '@/services/artifact';
+import * as DatasetService from '@/services/dataset';
+import * as ModelService from '@/services/model';
+import { AssetType } from '@/types/Types';
+import DatasetIcon from '@/assets/svg/icons/dataset.svg?component';
+
+const TIMEOUT_MS = 1000;
+
+const activeProject = shallowRef<IProject | null>(null);
+const allProjects = shallowRef<IProject[] | null>(null);
+
+/**
+ * Get the icon associated with an Asset
+ */
+const icons = new Map<string | AssetType, string | Component>([
+	[AssetType.Publications, 'file'],
+	[AssetType.Models, 'share-2'],
+	[AssetType.Datasets, DatasetIcon],
+	[AssetType.Simulations, 'settings'],
+	[AssetType.Code, 'code'],
+	[AssetType.Workflows, 'git-merge'],
+	['overview', 'layout']
+]);
+
+export function useProjects() {
+	async function getActiveProject(projectId: IProject['id']): Promise<IProject | null> {
+		if (projectId && !!projectId) {
+			activeProject.value = await ProjectService.get(projectId, true);
+		} else {
+			activeProject.value = null;
+		}
+		return activeProject.value;
+	}
+
+	async function getAllProjects(): Promise<IProject[]> {
+		allProjects.value = (await ProjectService.getAll()) as unknown as IProject[];
```

Comment:
Seems like this should just be making the call rather than using the project service.
Should probably remove project service all together if im understanding correctly

This goes for all of these i think but this is a good example

Reply:
Not sure if im making sense above. This section is just a duplication of the project service essentially.
Would be cleaner to have no duplication anywhere so if this getAll call needs to get updated we dont need to change it here as well as the service file

Reply:
I have a somewhat similar thought here, it feels a little bit like in some cases this composable is over-stepping its boundaries, by accessing other services such as DataService, ModelService, CodeService etc etc. 

I am not really sure what to feel about that but definitely need to watch out for getting into situations of circular dependencies, and potentially overloading the composable with too many responsibilities. 

I guess I am somewhat ambivalent here but curious to hear other peoples' thoughts here.

Reply:
It's not exactly a duplication of ProjectService, ProjectService is (mostly) pure API calls whereas this composable contains stateful logic.

Reply:
@mwdchang regarding boundaries, fair point; the reason why I am calling other services is because every time we, for example, call `CodeService.uploadCodeToProject`, we are also adding the code asset to the project, so as a matter of convenience it might make sense to call one function that does that rather than calling two functions. But I can see how keeping them separate might be preferable.

Reply:
I agree with @mwdchang, I don't think `useProject` should do that much.

i.e. If someone wants to save a Dataset and add it to a project, they should do:
```Javascript
const datasetId = useDataset.create();
useProject.addAsset(datasetId); // If the projectId is omitted, we use the active project
```